src/lock_file/package_identifier.rs

```diff
@@ -151,16 +152,32 @@ impl PypiPackageIdentifier {
             }
             // a pypi -> conda requirement on these versions are not supported
             uv_distribution_types::RequirementSource::Url { .. } => {
-                unreachable!("direct url requirement on conda package is not supported")
+                tracing::warn!(
+                    "PyPI requirement: {} as an url dependency is currently not supported as it is already selected as a conda package",
+                    consts::PYPI_PACKAGE_STYLE.apply_to(requirement.name.as_str())
+                );
+                Ok(false)
             }
             uv_distribution_types::RequirementSource::Git { .. } => {
-                unreachable!("git requirement on conda package is not supported")
+                tracing::warn!(
+                    "PyPI requirement: {} as a Git dependency is currently not supported as it is already selected as a conda package",
+                    consts::PYPI_PACKAGE_STYLE.apply_to(requirement.name.as_str())
+                );
+                Ok(false)
             }
             uv_distribution_types::RequirementSource::Path { .. } => {
-                unreachable!("path requirement on conda package is not supported")
+                tracing::warn!(
+                    "PyPI requirement: {} as a path dependency is currently not supported as it is already selected as a conda package",
+                    consts::PYPI_PACKAGE_STYLE.apply_to(requirement.name.as_str())
+                );
+                Ok(false)
             }
             uv_distribution_types::RequirementSource::Directory { .. } => {
-                unreachable!("directory requirement on conda package is not supported")
+                tracing::warn!(
+                    "PyPI requirement: {} as directory dependency is currently not supported as it is already selected as a conda package",
```

Comment:
```suggestion
                    "PyPI requirement: {} as directory dependency is currently not supported because it is already selected as a conda package",
```